### PR TITLE
Move misc from marble, rename and deprecate

### PR DESCRIPTION
**Pull request by pepe**
In the pursuit of moving my codes to spork, I have added many to the misc module. It ranges from simple mapping functions to PGP word generators.

The peg-grammar is the special addition I would like to pick for better investigation by the reviewer. I meant it as a grab bag of the peg rules and a simple mechanism to use them in your code. Presumably, we will add much more to this registry.

I also streamlined docstrings in the whole module with a simple rule: if it fits on one line, use " if it is not, use ``` on its line.

I also added prefixes to a couple of functions already in the module (with the old name as an alias, with a deprecation flag), which is out of scope for this commit but plays well with the new additions.

All the new functionality is tested and documented.

### CODE REVIEW

File: doc/misc.mdz [hunk 1]
```diff
@@ -60,3 +60,325 @@ x       y
...
+
+## caperr
+
+Captures the standart error output of the variadic body and returns it as
+a buffer.
+
+@codeblock[janet]```
+(misc/capout
+  (def p "HOHOHO")
+  (prin p))
```
Comment (l63l) [inline doc/misc.mdz:176]: needs edit after copy&paste

Comment (pepe) [reply doc/misc.mdz:176]: I have looked at this part and do not see any problem. May you please be more specific? I found another problem with a block on line 187 though :-D.

Comment (l63l) [reply doc/misc.mdz:175]: it is the example from capout copy&pasted to caperr and not:
```
(misc/caperr
  (def p "HOHOHO")
  (eprin p))
```

Comment (pepe) [reply doc/misc.mdz:175]: Oh, the programmer's blindness. Thank you very much!

**Comment by bakpakin** [comment]
LGTM

**Pull request merged**